agent/algorithm/binary/binary.go

```diff
@@ -40,28 +37,25 @@ func (b *binary) AddDataset(dataset string) {
 }
 
 func (b *binary) Run() ([]byte, error) {
-	defer os.Remove(b.algoFile)
+	if err := os.Mkdir(algorithm.ResultsDir, 0o755); err != nil {
+		return nil, fmt.Errorf("error creating results directory: %s", err.Error())
+	}
```

Comment:
let agent handle this since it is common for all algos, same with data directory